plugins/package-managers/spdx/src/funTest/kotlin/SpdxDocumentFileFunTest.kt

```diff
@@ -255,6 +256,24 @@ class SpdxDocumentFileFunTest : WordSpec({
                 Identifier("SpdxDocumentFile:OpenSSL Development Team:openssl:1.1.1g")
             )
         }
+
+        "collect issues for subprojects using illegal SPDX identifiers" {
+            val projectFile = projectDir.resolve("illegal-chars-external-refs/project-xyz.spdx.yml")
+            val subProjectFile = projectDir.resolve("illegal-chars-external-refs/illegal_chars/package.spdx.yml")
+            val definitionFiles = listOf(projectFile, subProjectFile)
+
+            val result = create("SpdxDocumentFile").resolveDependencies(definitionFiles, emptyMap())
+
+            val rootProject = result.projectResults[projectFile.absoluteFile]?.first()
+
+            rootProject.shouldNotBeNull()
+            rootProject.issues shouldHaveSize 1
+            rootProject.issues.first().message shouldContain Regex(
```

Comment:
I propose to instead use `shouldHaveSingleElement` with a boolean expression to match the Regex.